client/jetpack-cloud/sections/partner-portal/assign-license-step-progress/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { Gridicon } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { ReactElement } from 'react';
+import './style.scss';
+
+function getStepClassName( currentStep: number, step: number ): any {
+	switch ( true ) {
+		case currentStep === step:
+			return 'step-current';
+		case currentStep < step:
+			return 'step-next';
+		case currentStep > step:
+			return 'step-complete';
+	}
```

Comment:
While this is an interesting usage of a switch, this looks like language abuse to me 😆 
We can use normal if()s or even `classnames( { 'step-current': ..., ... }  )` as this is a rather convoluted way to achieve the same IMO.